packages/angular_devkit/build_angular/src/builders/dev-server/vite-server.ts

```diff
@@ -591,7 +595,7 @@ export async function setupServer(
             if (assetSourcePath !== undefined) {
               // The encoding needs to match what happens in the vite static middleware.
               // ref: https://github.com/vitejs/vite/blob/d4f13bd81468961c8c926438e815ab6b1c82735e/packages/vite/src/node/server/middlewares/static.ts#L163
-              req.url = `/@fs/${encodeURI(assetSourcePath)}`;
+              req.url = `${serverOptions.servePath ?? ''}/@fs/${encodeURI(assetSourcePath)}`;
```

Comment:
```suggestion
              req.url = `${server.config.base}@fs/${encodeURI(assetSourcePath)}`;
```
Can you try with the above?  Vite should normalize the input value and its also what it appears to be checking in the asset middleware.